homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,277 @@
+from homeassistant.core import callback
+from homeassistant.components.cover import (
+    DOMAIN, PLATFORM_SCHEMA, CoverDevice, ATTR_POSITION,
+    ATTR_CURRENT_POSITION, ATTR_TILT_POSITION, ATTR_CURRENT_TILT_POSITION,
+    SUPPORT_OPEN, SUPPORT_CLOSE, SUPPORT_STOP, SUPPORT_SET_POSITION,
+    SUPPORT_OPEN_TILT, SUPPORT_CLOSE_TILT,
+    SUPPORT_STOP_TILT, SUPPORT_SET_TILT_POSITION,
+    SERVICE_OPEN_COVER, SERVICE_CLOSE_COVER, SERVICE_SET_COVER_POSITION,
+    SERVICE_STOP_COVER, SERVICE_OPEN_COVER_TILT, SERVICE_CLOSE_COVER_TILT,
+    SERVICE_STOP_COVER_TILT, SERVICE_SET_COVER_TILT_POSITION)
+from homeassistant.const import (
+    ATTR_ASSUMED_STATE, ATTR_ENTITY_ID, ATTR_SUPPORTED_FEATURES,
+    CONF_ENTITIES, CONF_NAME, STATE_CLOSED)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change
+
+_LOGGER = logging.getLogger(__name__)
+
+TILT_FEATURES = SUPPORT_OPEN_TILT | SUPPORT_CLOSE_TILT | \
+                SUPPORT_STOP_TILT | SUPPORT_SET_TILT_POSITION
+
+KEY_OPEN_CLOSE = 'open_close'
+KEY_STOP = 'stop'
+KEY_POSITION = 'position'
+
+DEFAULT_NAME = 'Cover Group'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_ENTITIES): cv.entities_domain(DOMAIN),
+})
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the Group Cover platform."""
+    async_add_devices(
+        [CoverGroup(config[CONF_NAME], config[CONF_ENTITIES])])
+
+
+class CoverGroup(CoverDevice):
+    """Representation of a CoverGroup."""
+
+    def __init__(self, name, entities):
+        """Initialize a CoverGroup entity."""
+        self._name = name
+        self._tilt = False
+
+        self._is_closed = False
+        self._cover_position = 100
+        self._tilt_position = None
+        self._supported_features = 0
+        self._assumed_state = True
+
+        self._entities = entities
+        self._covers = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                        KEY_POSITION: set()}
+        self._tilts = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                       KEY_POSITION: set()}
+
+    @callback
+    def update_supported_features(self, entity_id, old_state, new_state,
+                                  update_state=True):
+        """Update dictionaries with supported features."""
+        if new_state is None:
+            for values in self._covers.values():
+                values.discard(entity_id)
+            for values in self._tilts.values():
+                values.discard(entity_id)
+        else:
+            features = new_state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+            # pylint: disable=expression-not-assigned
+            self._covers[KEY_OPEN_CLOSE].add(entity_id) \
+                if features & (SUPPORT_OPEN | SUPPORT_CLOSE) \
+                else self._covers[KEY_OPEN_CLOSE].discard(entity_id)
```

Comment:
Style comment: This is incredibly hard to read and has no benefits over the `if x:` `else:` style. Also I think it would get rid of the pylint issue.

```python
if features & (SUPPORT_OPEN | SUPPORT_CLOSE):
    self._covers[KEY_OPEN_CLOSE].add(entity_id)
else:
    self._covers[KEY_OPEN_CLOSE].discard(entity_id)
```

Reply:
:+1:

Reply:
I have changed it, but am also not that sure if it's actually better to read now.